apps/agentic-chat/src/hooks/useTools.ts

```diff
@@ -131,18 +131,12 @@ const useTools = () => {
       sellAmountCryptoPrecision: z
         .string()
         .describe('Amount to sell in human format, e.g. 1 for 1 ETH'),
-      fromAddress: z
-        .string()
-        .describe(
-          'The address the user is swapping from. Also referred to as "sell address", and should ALWAYS be gotten using the getAddress() tool beforehand'
-        ),
     }),
     execute: async ({
       chain,
       fromAsset,
       toAsset,
       sellAmountCryptoPrecision,
-      fromAddress,
```

Comment:
Somehow trying to get `fromAddress` carried over with qwen from e.g `getAccount()` didn't work proper - o4-mini just is smarter / larger context window to remember it, but we don't even need a `getAddress()` call, we simply need to use the from address at tool execution-time, which is available from the wagmi `WalletProvider` hook.